### PR TITLE
[lutron] Replace calls to methods deprecated in Java 11

### PR DESCRIPTION
**Pull request by bobadair**
Replaces calls to methods deprecated in Java 11.

### CODE REVIEW

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GreenModeHandler.java [hunk 1]
```diff
@@ -140,7 +141,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             } else if (command == OnOffType.OFF) {
                 greenMode(ACTION_STEP, 1);
             } else if (command instanceof Number) {
-                Integer step = new Integer(((Number) command).intValue());
+                Integer step = Integer.valueOf(((Number) command).intValue());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/GreenModeHandler.java:144]:
```suggestion
                Integer step = ((Number) command).intValue();
```

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java [hunk 2]
```diff
@@ -102,7 +102,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
         if (channelUID.getId().equals(CHANNEL_CLOCKMODE)) {
             if (command instanceof DecimalType) {
-                Integer mode = new Integer(((DecimalType) command).intValue());
+                Integer mode = Integer.valueOf(((DecimalType) command).intValue());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java:105]:
```suggestion
                Integer mode = ((DecimalType) command).intValue();
```

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java [hunk 3]
```diff
@@ -111,7 +111,7 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             }
         } else if (channelUID.getId().equals(CHANNEL_EXECEVENT)) {
             if (command instanceof DecimalType) {
-                Integer index = new Integer(((DecimalType) command).intValue());
+                Integer index = Integer.valueOf(((DecimalType) command).intValue());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java:114]:
```suggestion
                Integer index = ((DecimalType) command).intValue();
```

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java [hunk 4]
```diff
@@ -130,14 +130,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
             }
         } else if (channelUID.getId().equals(CHANNEL_ENABLEEVENT)) {
             if (command instanceof DecimalType) {
-                Integer index = new Integer(((DecimalType) command).intValue());
+                Integer index = Integer.valueOf(((DecimalType) command).intValue());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java:133]:
```suggestion
                Integer index = ((DecimalType) command).intValue();
```

File: bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java [hunk 5]
```diff
@@ -130,14 +130,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
                 timeclock(ACTION_SETEVENT, index, EVENT_ENABLE);
             } else {
                 logger.debug("Invalid command type for enableevent channnel");
             }
         } else if (channelUID.getId().equals(CHANNEL_DISABLEEVENT)) {
             if (command instanceof DecimalType) {
-                Integer index = new Integer(((DecimalType) command).intValue());
+                Integer index = Integer.valueOf(((DecimalType) command).intValue());
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/handler/TimeclockHandler.java:140]:
```suggestion
                Integer index = ((DecimalType) command).intValue();
```

**Review by cpmeister — CHANGES_REQUESTED** [review]
Might as well also take advantage of java's automatic boxing feature.

**Comment by bobadair** [comment]
> Might as well also take advantage of java's automatic boxing feature.

Yeah, I should have cleaned that up before.

**Review by cpmeister — APPROVED** [review]
LGTM

**Pull request merged**